base_tier_validation/models/tier_validation.py

```diff
@@ -472,15 +481,8 @@ def request_validation(self):
                     for td in tier_definitions:
                         if rec.evaluate_tier(td):
                             sequence += 1
-                            created_trs += tr_obj.create(
-                                {
-                                    "model": self._name,
-                                    "res_id": rec.id,
-                                    "definition_id": td.id,
-                                    "sequence": sequence,
-                                    "requested_by": self.env.uid,
-                                }
-                            )
+                            vals = rec._prepare_tier_review_vals(td, sequence)
+                            created_trs += tr_obj.create(vals)
```

Comment:
Please take the occasion to accumulate the dictionaries and create all the records at once.